commandapi-platforms/commandapi-velocity/commandapi-velocity-plugin/src/main/java/dev/jorel/commandapi/config/VelocityConfigurationAdapter.java

```diff
@@ -0,0 +1,153 @@
+package dev.jorel.commandapi.config;
+
+import org.spongepowered.configurate.CommentedConfigurationNode;
+import org.spongepowered.configurate.ConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+import org.spongepowered.configurate.yaml.NodeStyle;
+import org.spongepowered.configurate.yaml.YamlConfigurationLoader;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+public record VelocityConfigurationAdapter(YamlConfigurationLoader loader, CommentedConfigurationNode config, DefaultVelocityConfig defaultVelocityConfig) implements ConfigurationAdapter<ConfigurationNode> {
+
+	public static VelocityConfigurationAdapter createDummyInstance() {
+		return new VelocityConfigurationAdapter(null, null, null);
+	}
+
+	@Override
+	public void setValue(String key, Object value) {
+		try {
+			node(key).set(value);
+		} catch (SerializationException e) {
+			e.printStackTrace(System.err);
+		}
+	}
+
+	@Override
+	public void setComment(String key, String[] comment) {
+	}
+
+	@Override
+	public Object getValue(String key) {
+		try {
+			return node(key).get(Object.class);
+		} catch (SerializationException e) {
+			e.printStackTrace(System.err);
+			return null;
+		}
+	}
+
+	@Override
+	public String[] getComment(String key) {
+		CommentedConfigOption<?> option = defaultVelocityConfig.getAllOptions().get(key);
+		CommentedSection section = defaultVelocityConfig.getAllSections().get(key);
+		return option != null ? option.comment() : section.comment();
+	}
```

Comment:
I mean, the `ConfigGenerator` is written with comments in mind. This right here is essentially telling it to ignore comments since the check within the generator that checks if the comments are different will always fail since the default comment is compared with itself.